calculator_test.go

```diff
@@ -2,23 +2,167 @@ package calculator_test
 
 import (
 	"calculator"
+	"math"
+	"math/rand"
 	"testing"
 )
 
+type testCase struct {
```

Comment:
There's no problem with defining this struct at package level, but it's usual to define a test case struct within each test instead. That way, you can make it exactly what you need for the specific test, which will usually be slightly different each time.